src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/EnclaveDelegate.CngCryto.cs

```diff
@@ -0,0 +1,230 @@
+                        s_enclaveProviders[attestationProtocol] = hostGuardianServiceEnclaveProvider;
+                        sqlColumnEncryptionEnclaveProvider = s_enclaveProviders[attestationProtocol];
+                        break;
+
+#if ENCLAVE_SIMULATOR
+                    case SqlConnectionAttestationProtocol.SIM:
+                        SimulatorEnclaveProvider simulatorEnclaveProvider = new SimulatorEnclaveProvider();
+                        s_enclaveProviders[attestationProtocol] = (SqlColumnEncryptionEnclaveProvider)simulatorEnclaveProvider;
+                        sqlColumnEncryptionEnclaveProvider = s_enclaveProviders[attestationProtocol];
+                        break;
+#endif
+
+                    default:
+                        break;
+                }
+            }
+
+            if (sqlColumnEncryptionEnclaveProvider == null)
+            {
+                throw SQL.EnclaveProviderNotFound(enclaveType, ConvertAttestationProtocolToString(attestationProtocol));
+            }
+
+            return sqlColumnEncryptionEnclaveProvider;
+        }
+
+        /// <summary>
+        /// Generate the byte package that needs to be sent to the enclave
+        /// </summary>
+        /// <param name="attestationProtocol">attestation protocol</param>
+        /// <param name="keysToBeSentToEnclave">Keys to be sent to enclave</param>
+        /// <param name="queryText"></param>
+        /// <param name="enclaveType">enclave type</param>
+        /// <param name="enclaveSessionParameters">The set of parameters required for enclave session.</param>
+        /// <returns></returns>
+        internal EnclavePackage GenerateEnclavePackage(SqlConnectionAttestationProtocol attestationProtocol, Dictionary<int, SqlTceCipherInfoEntry> keysToBeSentToEnclave, string queryText, string enclaveType, EnclaveSessionParameters enclaveSessionParameters)
+        {
+            SqlEnclaveSession sqlEnclaveSession;
+            long counter;
+
+            try
+            {
+                GetEnclaveSession(
+                    attestationProtocol,
+                    enclaveType,
+                    enclaveSessionParameters,
+                    generateCustomData: false,
+                    sqlEnclaveSession: out sqlEnclaveSession,
+                    counter: out counter,
+                    customData: out _,
+                    customDataLength: out _,
+                    throwIfNull: true
+                );
+            }
+            catch (Exception e)
+            {
+                throw new RetryableEnclaveQueryExecutionException(e.Message, e);
+            }
+
+            List<ColumnEncryptionKeyInfo> decryptedKeysToBeSentToEnclave = GetDecryptedKeysToBeSentToEnclave(keysToBeSentToEnclave, enclaveSessionParameters.ServerName);
+            byte[] queryStringHashBytes = ComputeQueryStringHash(queryText);
```

Comment:
Since you are still keeping the CngCrypto version, can we still use the Sha256cng here? Currently, this ComputeQueryStringHash uses Sha256 across .net core and .net framework.

Reply:
I kept the Cng version because I don't think I should make decisions about standardizing on the cross platform version. I don't own the code so that isn't my decisions to make. I believe there are no reasons not to use only the cross platform version and that it would simplify the codebase. What do you think?

I can break this function out into CrossPlatform and CngCrypto versions if you like but there should be no difference other than the letters Cng at the end of the name as far as I know, is it worth it?

Reply:
Yea, I am just wondering about the benefit to use the Cng version in the first place. For this specific Sha256 case, I am OK with using the CrossPlatform version if they return the same hash value.

I can see the advantage to keep things standardized instead of maintaining two versions of code. The .net framework version is originally from the System.Data.SqlClient to keep the consistency. We will need to discuss this in the team before we entirely get rid of the whole Cng version stuff. For this PR, I think the current changes are good for now.

Reply:
My understanding is that on netfx the use of base classes will actually produce an implementaiton using the Cng classes. So it will be functionally identical. On non-windows it'll use ssl or mac crypto libraries through the netcore PAL layer. Contractually the runtime behaviour should be the same and the algorithms used are well defined and tested.

My assumption is that when the version using Cng was written it was a windows only codebase and there was no need to considering using the base classes.

Reply:
I'm confused here. We've decided to use the cross platform SHA256 class, so do we still need this CngCrypto file (typo in the name btw)?

Reply:
Has it been decided? If so I agree with the decision but I didn't know it had been made. 
I think it would be cleanest to merge this PR as it is and I will follow up with another to make only the changes required to move to managed for all builds. If I try to do it here you'll have to re-review everything.

Reply:
Sorry, yeah we discussed it internally. You're right, let's leave it for another pr.

Reply:
Well I suppose it's nice to be considered part of the team but I feel I'm missing a lot of meetings that I should be attending if that's the case 😁